Unhardcode music file types [Superseded]

penev92 opened the pull request:
Use the code from  #10069 to unhardcode music file types like that PR unhardcoded sound (effects, speech, and such) file types.
#10032 would have to take this into account as well, sorry @teees . At least it looks like it should be pretty simple.

pchote commented on OpenRA.Game/GameRules/MusicInfo.cs:
```diff
@@ -39,12 +41,18 @@ public void Load()
-					Length = (int)WavLoader.WaveLength(s);
-				else
-					Length = (int)AudLoader.SoundLength(s);
+				foreach (var loader in Game.ModData.SoundLoaders)
```
It would be much cleaner to identify the file type first, and then call GetLength for that type.

penev92 replied:
Changed it to expose a `CanParse(Stream)` method via the interface.
The alternative would be to do filename checks, with each parser providing a file extension, but that's not very reliable as it can easily fail on bogus files. (Would be much faster though, so may very well be worth it).

penev92 committed:
Streamline file verification in AudLoader and WavLoader

Also add CanParse() and GetLength() to ISoundLoader.

Update VocLoader (by teees)

penev92 committed:
Unhardcode sound file formats from MusicInfo

penev92 commented:
Needs to rebase on top of #10032 now...

teees commented:
referencing commit is meant to make it work with the merged vocloader.
Any other commits in that branch are to rework the `ISoundLoader` interface and make it similar to the other file format loaders.

penev92 commented:
Rebased and squashed the commit by @teees into the first commit. Thanks @teees !
But I cannot verify that Dune II still works.

pchote commented:
I still think that it would be cleaner to split out @teees's `ISoundFormat` changes from his music streaming branch, otherwise we will end up having to revert these changes again in the medium term.

penev92 commented:
Superceded by #10676.

pchote commented:
Now properly superseded by #10912.
